presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeIcebergTpcdsWithPositionalDeletes.java

```diff
@@ -0,0 +1,265 @@
+                        "                           FROM   date_dim " +
+                        "                           WHERE  d_date BETWEEN date'2000-05-20' AND date'2000-05-21') " +
+                        "       AND ws_sold_date_sk <= (SELECT Max(d_date_sk) " +
+                        "                               FROM   date_dim " +
+                        "                               WHERE  d_date BETWEEN date'2000-05-20' AND " +
+                        "                                                     date'2000-05-21') ",
+                0L);
+
+        assertUpdateExpected(session, "DELETE FROM web_sales " +
+                        "WHERE  ws_sold_date_sk >= (SELECT Min(d_date_sk) " +
+                        "                           FROM   date_dim " +
+                        "                           WHERE  d_date BETWEEN date'1999-09-18' AND date'1999-09-19') " +
+                        "       AND ws_sold_date_sk <= (SELECT Max(d_date_sk) " +
+                        "                               FROM   date_dim " +
+                        "                               WHERE  d_date BETWEEN date'1999-09-18' AND " +
+                        "                                                     date'1999-09-19') ",
+                22L);
+
+        assertUpdateExpected(session, "DELETE FROM web_sales " +
+                        "WHERE  ws_sold_date_sk >= (SELECT Min(d_date_sk) " +
+                        "                           FROM   date_dim " +
+                        "                           WHERE  d_date BETWEEN date'2002-11-12' AND date'2002-11-13') " +
+                        "       AND ws_sold_date_sk <= (SELECT Max(d_date_sk) " +
+                        "                               FROM   date_dim " +
+                        "                               WHERE  d_date BETWEEN date'2002-11-12' AND " +
+                        "                                                     date'2002-11-13') ",
+                42L);
+        deletedRowsMapBuilder.put("web_sales", 64L);
+        deletedRowsMap = deletedRowsMapBuilder.build();
+    }
+
+    private String getCountQuery(String tableName)
+    {
+        return "SELECT COUNT(*) FROM " + tableName;
+    }
+
+    private void verifyDeletes()
+    {
+        Session tpcdsConnSession = Session.builder(session)
+                .setCatalog("tpcds")
+                .setSchema("tiny")
+                .build();
+        for (Map.Entry<String, Long> entry : deletedRowsMap.entrySet()) {
+            String tableName = entry.getKey();
+            Long numDeletedRows = entry.getValue();
+            String countQuery = getCountQuery(tableName);
+            Long originalRowcount = (long) computeScalarExpected(tpcdsConnSession, countQuery);
+            Long postDeleteRowcount = (long) computeScalar(session, countQuery);
+            assertEquals(originalRowcount - postDeleteRowcount, numDeletedRows);
```

Comment:
Just verifying the number of result rows seems too weak. Can we add a expectedQueryRunner as a member of the TestPrestoNativeIcebergTpcdsQueriesParquetUsingThrift class and compare the results?

Reply:
We are running all TPCDS queries after the deletes, so the actual and expected result comparison happens there. This I added for simple verification prior to running the full query set.